app/models/task.py

```diff
@@ -1,5 +1,25 @@
 from app import db
 
-
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
```

Comment:
Nice approach creating this relationship to your Goal model. Why not use lazy here?